packages/calcite-components/src/components/table-header/table-header.tsx

```diff
@@ -0,0 +1,227 @@
+    setComponentLoaded(this);
+  }
+
+  connectedCallback(): void {
+    connectLocalized(this);
+    connectMessages(this);
+  }
+
+  disconnectedCallback(): void {
+    disconnectLocalized(this);
+    disconnectMessages(this);
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+  @Element() el: HTMLCalciteTableHeaderElement;
+
+  @State() defaultMessages: TableHeaderMessages;
+
+  @State() effectiveLocale = "";
+
+  @Watch("effectiveLocale")
+  effectiveLocaleChange(): void {
+    updateMessages(this, this.effectiveLocale);
+  }
+
+  private thEl: HTMLDivElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  /** Sets focus on the component. */
+  @Method()
+  async setFocus(): Promise<void> {
+    await componentFocusable(this);
+    this.thEl.focus();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private getScreenReaderText(): string {
+    const sharedText = `${this.selectedRowCountLocalized} ${this.messages.selected}`;
+    if (this.numberCell) {
+      return this.messages.rowNumber;
+    } else if (this.selectionMode === "single") {
+      return `${this.messages.selectionColumn}. ${sharedText}`;
+    } else if (this.bodyRowCount === this.selectedRowCount) {
+      return `${this.messages.selectionColumn}. ${this.messages.all} ${sharedText} ${this.messages.keyboardDeselectAll}`;
+    } else {
+      return `${this.messages.selectionColumn}. ${sharedText} ${this.messages.keyboardSelectAll}`;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  //--------------------------------------------------------------------------
+
+  render(): VNode {
+    const scope = this.rowSpan
+      ? "rowgroup"
+      : this.colSpan
+      ? "colgroup"
+      : this.parentRowType === "body"
+      ? "row"
+      : "col";
+
+    const allSelected = this.selectedRowCount === this.bodyRowCount;
+    const selectionIcon = allSelected ? "check-square-f" : "check-square";
+
+    return (
+      <Host>
+        <th
+          aria-colindex={this.parentRowType !== "body" ? this.positionInRow : ""}
+          class={{
+            [CSS.bodyRow]: this.parentRowType === "body",
+            [CSS.numberCell]: this.numberCell,
+            [CSS.selectionCell]: this.selectionCell,
+            [CSS.multipleSelectionCell]: this.selectionMode === "multiple",
+          }}
+          colSpan={this.colSpan}
+          role="columnheader"
+          rowSpan={this.rowSpan}
+          scope={scope}
+          tabindex={0}
```

Comment:
should be `tabIndex={0}`. Can you verify the all lowercase works here?

Reply:
Updated - but they both seem to work - maybe we need a convention, seems mixed across our components.

Reply:
yeah maybe an eslint rule if we can. I know in maquette if you do this it won't work because it expects a number for `tabIndex` (camelcase) and a string for `tabindex` (lowercase)

Reply:
Added follow-up issue https://github.com/Esri/calcite-design-system/issues/7631